### PR TITLE
feat(iOS/paper): percentage support in translate

### PR DESCRIPTION
**Pull request by intergalacticspacehighway**
## Summary:
This PR adds percentage support in translate properties for old arch iOS. Isolating this PR for easier reviews.

## Changelog:
[IOS] [ADDED] - Percentage support in translate in old arch.

## Test Plan:
- Checkout TransformExample.js -> Translate percentage example. 
- Added a simple test in `processTransform-test.js`

Related PRs - https://github.com/facebook/react-native/pull/43193, https://github.com/facebook/react-native/pull/43192

### CODE REVIEW

File: packages/react-native/React/Views/RCTConvert+Transform.m [hunk 1]
```diff
@@ -144,6 +144,122 @@ + (CATransform3D)CATransform3D:(id)json
   return transform;
 }
 
++ (CATransform3D)CATransform3D:(id)json withWidth:(CGFloat)width height:(CGFloat)height {
```
Comment (javache) [inline packages/react-native/React/Views/RCTConvert+Transform.m:147]: Can we have `[RCTConvert CATransform3D:]` call `[RCTConvert CATransform3D:json withWidth:NaN height:NaN]` or something similar? We don't want to be duplicating this code.

Any caller to the other CATransform should error or fallback to zero when width/height are not provided.

File: packages/react-native/React/Views/UIView+React.h [hunk 2]
```diff
@@ -115,7 +115,7 @@ typedef struct {
  * The anchorPoint property doesn't work in the same way as on web - updating it updates the frame.
  * To work around this, we take both the transform and the transform-origin, and compute it ourselves
  */
-@property (nonatomic, assign) CATransform3D reactTransform;
+@property (nonatomic, copy) NSArray* reactTransform;
```
Comment (javache) [inline packages/react-native/React/Views/UIView+React.h:118]: Would `retain` be sufficient here if we make sure callers do not hold on to the transform?

The implementation's already not copying anyway, so this is currently not true :)

File: packages/react-native/React/Views/UIView+React.m [hunk 3]
```diff
@@ -205,22 +206,23 @@ - (void)reactSetFrame:(CGRect)frame
   self.bounds = bounds;
 
   id transformOrigin = objc_getAssociatedObject(self, @selector(reactTransformOrigin));
-  if (transformOrigin) {
+  id transform = objc_getAssociatedObject(self, @selector(reactTransform));
+  if (transformOrigin || transform) {
     updateTransform(self);
   }
```
Comment (javache) [inline packages/react-native/React/Views/UIView+React.m:212]: There's going to be a performance cost to parsing this more frequently (I guess anytime the frame changes right now?). Could we limit the updates to only when the bounds have changed?

Comment (intergalacticspacehighway) [reply packages/react-native/React/Views/UIView+React.m]: Thanks for catching, updated.

File: packages/react-native/React/Views/UIView+React.h (hunk 2 above)
Comment (intergalacticspacehighway) [reply packages/react-native/React/Views/UIView+React.h]: Yup, fixed!

File: packages/react-native/React/Views/RCTConvert+Transform.m (hunk 1 above)
Comment (intergalacticspacehighway) [reply packages/react-native/React/Views/RCTConvert+Transform.m:54]: Makes sense, thanks. Added an error log and returned 0 when height/width is not provided for percentage values.

**Pull request closed without merging**